src/views/FlexExampleAppHomeView.mxml

```diff
@@ -1,204 +1,129 @@
-				trace("[INCENTIVIZED] DID_HIDE");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.DID_FAIL, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] DID_FAIL");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.ON_AUDIO_STARTED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] ON_AUDIO_STARTED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.ON_AUDIO_FINISHED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] ON_AUDIO_FINISHED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_AVAILABLE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_AVAILABLE");
-				if (IncentivizedAd.getInstance().isAvailable()) {
-					trace("Incentivized Ad is available.");
-				} else {
-					trace("Incentivized Ad is not available");
-				}
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.FETCH_FAILED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] FETCH_FAILED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_COMPLETED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_COMPLETED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_INCOMPLETE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_INCOMPLETE");
-				return null;
-			});
-			
-		]]>
+        import com.heyzap.sdk.ads.BannerAd;
+        import com.heyzap.sdk.ads.HeyzapAds;
+        import com.heyzap.sdk.ads.IncentivizedAd;
+        import com.heyzap.sdk.ads.InterstitialAd;
+        import com.heyzap.sdk.ads.VideoAd;
+
+        import model.AppModel;
+
+        [Bindable]
+        public var appModel:AppModel;
+
+        private var currentActionObject:*;
+
+        override public function set data(value:Object):void {
+            appModel = value as AppModel;
+        }
+
+        private function onCreationComplete():void {
+            currentActionObject = InterstitialAd.getInstance();
+        }
+
+        private function getEnteredAdTagOrNull():String{
+            if (!adTagTextInput.text) return null;
+            else return adTagTextInput.text;
+        }
+
+        private function onFetchButtonClick():void {
+            currentActionObject.fetch(getEnteredAdTagOrNull());
+        }
+
+        private function onHideButtonClick():void {
+            currentActionObject.hide();
+            appModel.bannerIsShown = false;
+        }
+
+        private function onShowButtonClick():void {
+            if (currentActionObject instanceof  BannerAd) {
+                currentActionObject.show(BannerAd.POSITION_BOTTOM,getEnteredAdTagOrNull()); //todo add position
+                appModel.bannerIsShown = true;
```

Comment:
this is not strictly true. banner ads can fail to show, in which case the test app will fall into an inconsistent state. You could listen for the `loaded` callback and set this to true there.

(Or, even better, don't have state. let all the buttons stay enabled, so they strictly make function calls, whether or not they're appropriate calls to make. let the user determine if it's the right time to call those methods, by showing the callbacks in the console view you have)